code/modules/telesci/dses.dm

```diff
@@ -0,0 +1,303 @@
+		if(module_in_list("dirGET"))
+			t += "<BR>Last logged direction: [last_direction]"
+		if(module_in_list("getDST"))
+			t += "<BR>Distance from previous ping: [last_distance]"
+		if(module_in_list("crdLOG") && positive_locations.len)
+			t += "<BR><HR>Logged successful pings</HR>"
+			var/current_loc_itt
+			for(var/datum/dses_find/D in positive_locations)
+				current_loc_itt ++
+				t += {"<BR>\the [D.name] in [D.location_name] at [D.x-WORLD_X_OFFSET[D.z]] - [D.y-WORLD_Y_OFFSET[D.z]] - [D.z]
+				 : <A href='?src=\ref[src];wipe=[current_loc_itt]'>Wipe log</A>"}
+			t += "<BR><A href='?src=\ref[src];wipe=0'>Wipe All</A>"
+		t += "<BR><B><HR>Modules installed</HR></B>"
+		if(module_list.len)
+			var/current_module_num = 0
+			for(var/obj/item/weapon/dses_module/D in module_list)
+				current_module_num ++
+				t += "<BR>Module: [D.module_name] <A href='?src=\ref[src];eject=[current_module_num]'>Eject Module</A>"
+		else
+			t += "<BR>No modules detected"
+		t += "<BR>Module slots [module_list.len]/[module_limit]"
+		t += "<BR><A href='?src=\ref[src];eject_cell=1'>Eject Cell</A>"
+	else
+		t += "<B>No cell detected.</B>"
+	var/datum/browser/popup = new(user, "\ref[src]", name, 300, 450)
+	popup.set_content(t)
+	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+
+/obj/item/device/dses/Topic(href, href_list)
+	..()
+
+	usr.set_machine(src)
+
+	if (usr.get_active_hand() != src || usr.isUnconscious())
+		to_chat(usr, "<span class = 'caution'>You need to have \the [src] in your hand to do that!</span>")
+		return
+
+	if(href_list["pulse"])
+		if(C.charge < pulse_cost)
+			to_chat(usr, "<span class = 'caution'>\The [C] has insufficient charge to support another pulse.</span>")
+			return
+
+		handle_pulse(usr)
+		. = 1
+
+	if(href_list["eject"])
+		var/index = text2num(href_list["eject"])
+		if(index && index <= module_list.len)
+			var/obj/item/weapon/dses_module/D = module_list[index]
+			uninstall_module(D, usr)
+			D = null
+			. = 1
+
+	if(href_list["wipe"])
+		var/index = text2num(href_list["wipe"])
+		if(index && index <= positive_locations.len)
+			var/datum/dses_find/D = positive_locations[index]
+			positive_locations -= D
+			qdel(D)
+		else
+			for(var/datum/dses_find/D in positive_locations)
+				qdel(D)
+			positive_locations = list()
+		. = 1
+
+	if(href_list["tog_pulse"])
+		auto_pulse = !auto_pulse
+		. = 1
+
+	if(href_list["eject_cell"])
+		if(C)
+			usr.put_in_hands(C)
+			C = null
+			usr << browse(null, "window=\ref[src]")
+			return
+
+	if(.)
+		updateSelfDialog(usr)
+
+
+/obj/item/device/dses/process()
+	if(auto_pulse && last_pulse <= world.time+pulse_between)
+		handle_pulse()
+
+/obj/item/device/dses/proc/handle_pulse(mob/user)
+	if(last_pulse <= world.time+pulse_between)
+		var/atom/detected = pulse()
+
+		if(!detected)
+			if(user)
+				to_chat(user, "<span class = 'warning'>No structures detected.</span>")
+			playsound(src, 'sound/machines/buzz-sigh.ogg', 50, 1, gas_modified = 0)
```

Comment:
Alright, major problem, playsound_local and crow's suggestion both require a mob argument, I was kinda hoping of having it ping in pocket when it's set to autoping.

Maybe have an argument where if there is no user, it plays to all mobs on the turf?

Gonna try that and see how it goes